src/Features/Core/Portable/Workspace/BackgroundCompiler.cs

```diff
@@ -1,196 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
-using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Internal.Log;
-using Microsoft.CodeAnalysis.SolutionCrawler;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Host
-{
-    internal sealed class BackgroundCompiler : IDisposable
-    {
-        private Workspace _workspace;
-        private readonly IDocumentTrackingService _documentTrackingService;
-        private readonly TaskQueue _taskQueue;
-
-        [SuppressMessage("CodeQuality", "IDE0052:Remove unread private members", Justification = "Used to keep a strong reference to the built compilations so they are not GC'd")]
-        private Compilation?[]? _mostRecentCompilations;
```

Comment:
Isn't the purpose of this to have `RecoverableValueSource` compilation instances rooted here?

I'd imagine that without this code, the Compilation instances would likely be GC'd constantly.

See https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/SolutionState.CompilationTracker.CompilationTrackerState.cs,117